### DISCUSSION TITLE
Add version to ArduinoOTA-call

### DISCUSSION THREAD

**Discussion by dennisfabri** [arendst/Tasmota#11243] [Ideas]
The ArduinoOTA library supports sending a version string with the update request via http. This string can be used to determine the correct update firmware on the server side.

This is an example of the changes necessary (in support_tasmota.ino) to send a meaningfull version with the update request:
```
char version[40];
strcat_P(version, TasmotaGlobal.image_name);
strcat_P(version, "-");
strcat_P(version, TasmotaGlobal.version);
WiFiClient OTAclient;
ota_result = (HTTP_UPDATE_FAILED != ESPhttpUpdate.update(OTAclient, TasmotaGlobal.mqtt_data, version));
```

With these changes the following is possible:

1. The client requests a firmware update from http://myserver/ota.php: Version (sensors)-9.1.0
2. The server 
    2.1. stores the imagename for the chipid
    2.2. serves a fitting new minimal image: Version (minimal)-9.3.1
3. The client installs the update (minimal image)
4. The client requests a firmware update from http://myserver/ota.php
5. The server
    5.1. get a request from a minimal image and knows the chipid from the previous call
    5.2. serves the fitting new image: Version (sensors)-9.3.1

Possible advantages:
1. All your device updates could be served from the exact same url.
2. You could simply update your devices without having to double check if you are installing the correct image.
2. Migration paths from older versions could be automatically followed (only serverside code required): v1.0.11  🔀  v3.9.22  🔀  v4.2.0  🔀  v5.14.0  🔀  v6.7.1  🔀  v7.2.0  🔀  v8.5.1  🔀  v9.1 🔀 current version. Which means you would only need to hit the update button until you are at the newest version.
3. The server side code could possibly be integrated into TasmoAdmin.

I would provide a pull request for the changes, but I would like beforehand if any major changes should be made to the code above.

**Comment by arendst** [comment]
Just implemented
